static/src/js/components/JsonFileUploadModal/JsonFileUploadModal.jsx

```diff
@@ -0,0 +1,198 @@
+
+  /**
+   * Callback function for handling file drop
+   * @param {File[]} acceptedFiles - Array of accepted files
+   */
+  const onDrop = useCallback((acceptedFiles) => {
+    if (acceptedFiles && acceptedFiles.length > 0) {
+      setFile(acceptedFiles[0])
+      setErrors([])
+    }
+  }, [])
+
+  const {
+    getRootProps, getInputProps
+  } = useDropzone({
+    onDrop,
+    accept: {
+      'application/json': ['.json'],
+      'text/plain': ['.json']
+    },
+    multiple: false
+  })
+
+  useEffect(() => {
+    if (!show) {
+      setFile(null)
+      setErrors([])
+    }
+  }, [show])
+
+  /**
+   * Handles the file upload and validation process
+   */
+  const handleUpload = () => {
+    if (!file) {
+      setErrors([{
+        id: generateId(),
+        message: 'Please select a file to upload.'
+      }])
+
+      return
+    }
+
+    const reader = new FileReader()
+    reader.onload = (e) => {
+      try {
+        const jsonData = JSON.parse(e.target.result)
+        const validationResult = validateJson({
+          jsonData,
+          schema
+        })
+        if (validationResult.errors) {
+          setErrors(validationResult.errors.map((error) => ({
+            id: generateId(),
+            message: error
+          })))
+        } else {
+          upload(jsonData)
+          toggleModal(false)
+        }
+      } catch (error) {
+        setErrors([{
+          id: generateId(),
+          message: 'Invalid JSON file. Please upload a valid JSON file.'
+        }])
+      }
+    }
+
+    reader.readAsText(file)
+  }
+
+  return (
+    <CustomModal
+      show={show}
+      toggleModal={toggleModal}
+      header="Upload Draft JSON"
+      message={
+        (
+          <>
+            <div
+              className="file-upload-area"
+              // eslint-disable-next-line react/jsx-props-no-spreading
+              {...getRootProps()}
+            >
+              <input
+                type="file"
+                aria-label="Upload JSON file"
+                accept="application/json,text/plain"
+                onChange={getInputProps().onChange}
+                onBlur={getInputProps().onBlur}
+                onClick={getInputProps().onClick}
+                ref={getInputProps().ref}
+                style={getInputProps().style}
+              />
+              <p>Drop a JSON file here or click to upload</p>
+            </div>
+            {
+              file && (
+                <div className="alert alert-info mt-3">
+                  <FaFile className="me-2" />
+                  Selected file:
+                  {' '}
+                  {file.name}
+                </div>
+              )
+            }
+            {
+              errors.length > 0 && (
+                <div className="alert alert-danger">
+                  <ul className="mb-0">
+                    {
+                      errors.map((error) => (
+                        <li key={error.id}>{error.message}</li>
```

Comment:
Instead of using `error.id` you could also JSON.stringify(error), that is the other convention I see for giving the `li` a unique key.   But I think either way works.

Reply:
Agree, will keep the id then.